File: LeetCode/[M] - 2352. Equal Row and Column Pairs/solution.py
class Solution:
    def equalPairs(self, grid: list[list[int]]) -> int:
        rowsCnt = {}
        for row in grid:
            row = tuple(row)
            rowsCnt[row] = rowsCnt.get(row, 0) + 1
        res = 0
        for i in range(len(grid)):
            column = []
            for j in range(len(grid)):
                column.append(grid[j][i])
            res += rowsCnt.get(tuple(column), 0)
        return res

File: LeetCode/[M] - 2352. Equal Row and Column Pairs/solution.go
package main

func equalPairs(grid [][]int) int {
	rowsCnt := make(map[[200]int]int)
	row := [200]int{}
	for _, r := range grid {
		copy(row[:], r)
		rowsCnt[row]++
	}
	res := 0
	column := [200]int{}
	for i := 0; i < len(grid); i++ {
		for j := 0; j < len(grid); j++ {
			column[j] = grid[j][i]
		}
		res += rowsCnt[column]
	}
	return res
}
